model/version_modification.go

```diff
@@ -74,7 +77,16 @@ func ModifyVersion(version Version, user user.DBUser, modifications VersionModif
 				return http.StatusUnauthorized, errors.Errorf("insufficient access to set priority %d, can only set priority less than or equal to %d", modifications.Priority, evergreen.MaxTaskPriority)
 			}
 		}
-		if err := SetVersionsPriority([]string{version.Id}, modifications.Priority, user.Id); err != nil {
+		versionIds := []string{version.Id}
+		if evergreen.IsPatchRequester(version.Requester) {
```

Comment:
Since `GetFinalizedChildPatchIdsForPatch` will no-op if it's not a patch we can remove this if statement

Reply:
well it does no op but it's still a waste of a query, if we do know the requester ahead of time it makes sense to not bother with it

Reply:
Ah, I thought the `IsValidId` would make it no-op before the DB query was made but perhaps not